code/addons/interactions/src/Panel.tsx

```diff
@@ -43,16 +43,19 @@ export const getInteractions = ({
   const callsById = new Map<Call['id'], Call>();
   const childCallMap = new Map<Call['id'], Call['id'][]>();
   return log
-    .filter(({ callId, parentId }) => {
-      if (!parentId) return true;
-      childCallMap.set(parentId, (childCallMap.get(parentId) || []).concat(callId));
-      return !collapsed.has(parentId);
+    .filter(({ callId, ancestors }) => {
+      let visible = true;
+      ancestors.forEach((ancestor) => {
+        if (collapsed.has(ancestor)) visible = false;
+        childCallMap.set(ancestor, (childCallMap.get(ancestor) || []).concat(callId));
+      });
+      return visible;
     })
     .map(({ callId, status }) => ({ ...calls.get(callId), status } as Call))
     .map<Interaction>((call) => {
       const status =
         call.status === CallStates.ERROR &&
-        callsById.get(call.parentId)?.status === CallStates.ACTIVE
+        callsById.get(call.ancestors.slice(-1)[0])?.status === CallStates.ACTIVE
```

Comment:
This looks a little dangerous. Can it be extracted to a local constant with some sort of safe check + default value?

Reply:
`call.ancestors` is guaranteed to be an array (by TypeScript). Slicing an empty array always yields an empty array. Taking the first element of an empty array returns `undefined`, which we then check for using optional chaining. I don't see the problem unless you see a reason for `ancestors` to ever be null or undefined?